### DISCUSSION TITLE
Bundle variants quantity

### DISCUSSION THREAD

**Discussion by dhrpieterjan** [swellstores/community#167] [Help] [answered]
Hi community,

We would like to create a bundle that has different variants, Small, Medium, and Large. In the small bundle, there should be 2 products, for the medium bundle, we would like to increase the quantity of that product, same for the large bundle. 
Right now it works to create these 3 variants, but we don't know how we can increase te quantity of the products in the bundle. 

Does someone have any experience with this? 
Thank you!

**Comment by greph** [comment]
Hey @dhrpieterjan , are you referring to the inventory quantity on bundle variants? If so, it's fairly easy to setup: simply setup the options/variants and then enable inventory tracking on the product. You can then create inventory adjustments on a per-variant basis.

**Reply by dhrpieterjan** [reply]
Hey @greph , thx for your reply. We are more looking for a way to specify how many items are in a bundle, so for example:

Small week box = 1 bread + 1 baguette
Medium week box = 2 bread + 2 baguette
Large week box = 3 bread + 3 baguette

So we set up the bundle to have 3 variants (small, medium, and large) but we can not change the amount of the bundle items per variant...
Do you see what I mean? It is not logical to have 3 size variants of a bundle, all containing the same amount of items 🙃. 

We are working with a custom front-end so I need to get the number of items for a selected bundle variant in order to show this properly to the clients. Do you think we can do this via the inventory tracking?

**Reply by logeshswell** [reply, marked as answer]
One way you could do this is, adding a custom nested field inside `product.variants` as follows:

* Collection field (name: `products`)
  * Collections field (name: `bundler`)
    * Lookup field (name: `product_id` looking up products model)
    * Number field (name: `quantity`)

```ts
{
  id: 'product_id',
  variants: [
    {
      id: 'large_variant_id',
      name: 'large',
      products: [
        [
          'bread_product_id', // product_id
          3 // quantity
        ],
        ['baguette_product_id', 3] // bundler
      ]
    }
  ]
}
```

While adding the cart, you can just pass `product.id` and `product.variants.id`.

Otherwise, I would suggest you create 3 separate bundle products with different names for easier implementation. You can use the category field to group 3 different products together.
